src/reporter.ts

```diff
@@ -159,52 +152,26 @@ export default class Reporter {
     await this.testRunsApi.create(runs);
   }
 
-  async uploadAssets(
-    jobId: string | undefined,
-    video: string | null,
-    consoleLogContent: string,
-    screenshots: string[],
-    testReport: TestRun,
-  ) {
-    const assets = [];
-
-    // Since reporting is made by spec, there is only one video to upload.
-    if (video) {
-      assets.push({
-        data: fs.createReadStream(video),
-        filename: VIDEO_FILENAME,
-      });
-    }
-
-    // Add generated console.log
-    const logReadable = new stream.Readable();
-    logReadable.push(consoleLogContent);
-    logReadable.push(null);
-
-    const reportReadable = new stream.Readable();
-    reportReadable.push(testReport.stringify());
-    reportReadable.push(null);
-
-    assets.push(
-      {
-        data: logReadable,
-        filename: 'console.log',
-      },
-      {
-        data: reportReadable,
-        filename: 'sauce-test-report.json',
-      },
-    );
+  /**
+   * Converts a string into a readable stream.
+   * This method creates a new readable stream instance, pushes the provided data into it,
+   * and then signals the end of the stream.
+   *
+   * @param {string} data - The string data to be converted into a stream.
+   * @returns {stream.Readable} A readable stream containing the provided data.
+   */
+  strToReadableStream(data: string): stream.Readable {
```

Comment:
Why specific to strings? Doesn't need to be that restrictive. The underlying stream supports `any` and you're not actually doing anything with the string here aside from passing it into the stream.
```suggestion
  ReadableStream(data: unknown): stream.Readable {
```